src/Access/Common/AccessType.h

```diff
@@ -88,6 +88,8 @@ enum class AccessType
     M(CREATE_DICTIONARY, "", DICTIONARY, CREATE) /* allows to execute {CREATE|ATTACH} DICTIONARY */\
     M(CREATE_TEMPORARY_TABLE, "", GLOBAL, CREATE) /* allows to create and manipulate temporary tables;
                                                      implicitly enabled by the grant CREATE_TABLE on any table */ \
+    M(CREATE_ARBITRARY_TEMPORARY_TABLE, "", GLOBAL, CREATE)  /* allows to create  and manipulate temporary tables
+                                                                with arbitrary table engine */\
```

Comment:
OK. I made CREATE_TEMPORARY_TABLE a child of CREATE_ARBITRARY_TEMPORARY_TABLE

Reply:
@tavplubix 
I am not sure about parent_group_name parameter for:

```
M(CREATE_ARBITRARY_TEMPORARY_TABLE, "", GLOBAL, CREATE_TABLE) 
M(CREATE_TEMPORARY_TABLE, "", GLOBAL, CREATE_ARBITRARY_TEMPORARY_TABLE)
```

Because right now there are many Fast Test problems related to grants.

For example, I manually run:

```
SELECT dummy FROM remote('127.0.0.{1,2}', system, one)
```
And I get:
> Code: 497. DB::Exception: Received from localhost:9000. DB::Exception: default: Not enough privileges. To execute this query it's necessary to have grant REMOTE ON *.*. (ACCESS_DENIED)

My question was:

Why in master we have:
```
M(CREATE_TEMPORARY_TABLE, "", GLOBAL, CREATE)
```

and not:
```
M(CREATE_TEMPORARY_TABLE, "", GLOBAL, CREATE_TABLE)
```

???

In my point of view the 'hierarchy' is controlled by this code:

```
if (create.storage && create.storage->engine && create.storage->engine->name != "Memory")
    required_access.emplace_back(AccessType::CREATE_ARBITRARY_TEMPORARY_TABLE);
else
    required_access.emplace_back(AccessType::CREATE_TEMPORARY_TABLE);
```

And we should probably keep:

```
M(CREATE_ARBITRARY_TEMPORARY_TABLE, "", GLOBAL, CREATE) 
M(CREATE_TEMPORARY_TABLE, "", GLOBAL, CREATE)
```

What do you think?

Reply:
Maybe this is because node_type = GLOBAL. I am not sure.

Reply:
After some debugging, I found out that the declaration of a parent type before its child breaks access types in an obscure way. It was really unobvious, sorry if you wasted some on that. This patch should fix tests: https://gist.github.com/tavplubix/95571b8e9e15c233f30b6ac928fbbc19

Reply:
Thank you for the patch. I also noticed that the parent declaration goes after child's declaration in AccesType.h. But I could not finally conclude that this was the problem. I applied the patch and pushed my changes.